### PR TITLE
[test] Reorganize resources of prog. environment tests

### PR DESCRIPTION
**Pull request by jgphpc**
Will fix [UES-899](https://jira.cscs.ch/browse/UES-899)
tested on dom

@teojgo @ajocksch 
openacc_checks.py will fail with latest PrgEnv-cray (openacc no longer supported)

### CODE REVIEW

File: cscs-checks/prgenv/affinity_check.py [hunk 1]
```diff
@@ -17,6 +18,7 @@ def __init__(self, variant):
         # The github URL can not be specifid as `self.sourcedir` as that
         # would prevent the src folder from being copied to stage which is
         # necessary since these tests need files from it.
+        self.sourcesdir += os.path.join('/affinity_ref')
```
Comment (vkarak) [inline cscs-checks/prgenv/affinity_check.py:21]:
```suggestion
        self.sourcesdir = os.path.join(self.sourcesdir, 'affinity_ref')
```

And similarly in the rest.

File: cscs-checks/prgenv/helloworld.py [hunk 2]
```diff
@@ -121,6 +122,7 @@ class HelloWorldTestSerial(HelloWorldBaseTest):
     def __init__(self, lang, linkage):
         super().__init__('serial', lang, linkage)
         self.valid_systems += ['kesch:pn', 'arolla:pn', 'tsa:pn']
+        self.sourcesdir += os.path.join('/serial', lang)
```
Comment (vkarak) [inline cscs-checks/prgenv/helloworld.py:125]: I don't see the reason of having directories per language. I think it's too much. We only have a single file per language.

Comment (vkarak) [reply cscs-checks/prgenv/helloworld.py:125]: This comment applies everywhere.

File: cscs-checks/prgenv/opencl_check.py [hunk 3]
```diff
@@ -19,6 +19,7 @@ def __init__(self):
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
         self.modules = ['craype-accel-nvidia60']
         self.build_system = 'Make'
+        self.sourcesdir += os.path.join('/opencl')
         self.num_gpus_per_node = 1
         self.executable = 'vecAdd_opencl'
```
Comment (vkarak) [inline cscs-checks/prgenv/opencl_check.py:24]: If you move it to a specific directory, perhaps you don't need the `_opencl` suffix anymore.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/prgenv/opencl_check.py (hunk 3 above)
Comment (jgphpc) [reply cscs-checks/prgenv/opencl_check.py:25]: done

File: cscs-checks/prgenv/helloworld.py (hunk 2 above)
Comment (jgphpc) [reply cscs-checks/prgenv/helloworld.py]: done

File: cscs-checks/prgenv/affinity_check.py (hunk 1 above)
Comment (jgphpc) [reply cscs-checks/prgenv/affinity_check.py]: done

File: cscs-checks/prgenv/affinity_check.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ def __init__(self, variant):
         # The github URL can not be specifid as `self.sourcedir` as that
         # would prevent the src folder from being copied to stage which is
         # necessary since these tests need files from it.
-        self.sourcesdir += os.path.join('/affinity_ref')
+        self.sourcesdir += os.path.join('affinity_ref')
```
Comment (vkarak) [inline cscs-checks/prgenv/affinity_check.py:21]: I don't think this is correct.

File: cscs-checks/prgenv/helloworld.py [hunk 5]
```diff
@@ -6,6 +6,7 @@
 from datetime import datetime
 
 import os
+import re
```
Comment (vkarak) [inline cscs-checks/prgenv/helloworld.py:9]: Is this needed?

File: cscs-checks/prgenv/src/opencl/Makefile [hunk 6]
```diff
@@ -1,5 +1,5 @@
-vecAdd_opencl: vecAdd_opencl.o
+vecAdd_opencl: vecAdd.o
```
Comment (vkarak) [inline cscs-checks/prgenv/src/opencl/Makefile:1]: I meant to change the name of the final executable in my other comment.

**Review by vkarak — CHANGES_REQUESTED** [review]
The failure in the CI is because the affinity test is broken. Check my comment there. Apart from this, I have only a couple of minor comments.

File: cscs-checks/prgenv/helloworld.py (hunk 5 above)
Comment (jgphpc) [reply cscs-checks/prgenv/helloworld.py:9]: yes:
> git blame -L 94,94 helloworld.py 
```
d29909cfd (mkr 2020-06-25 12:55:56 +0200 94)         
envname = re.sub(r'(PrgEnv-\w+).*', lambda m: m.group(1),
```

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**